ethcore/light/src/net/mod.rs

```diff
@@ -818,7 +825,11 @@ impl LightProtocol {
 		}
 
 		let remote_flow = flow_params.map(|params| (params.create_credits(), params));
-		let local_flow = self.flow_params.read().clone();
+		let local_flow = if io.is_reserved_peer(*peer) {
+			self.free_flow_params.clone()
```

Comment:
Why not always return `FlowParams::free()` instead of adding the `free_flow_params` field?

Reply:
In https://github.com/paritytech/parity-ethereum/pull/9448/files#diff-114b98222286628ef4252b75052c348bR712 we only need a reference to `FlowParams::free()`, so while for L829 it's the same, on that one it saves one `FlowParams` stack allocation.

I don't think that's a big deal though so I'm happy to change it to always use `FlowParams::free()` directly.